lib/experimental/class-theme-json-webfonts-handler.php

```diff
@@ -0,0 +1,537 @@
+
+					return false;
+				}
+
+				// Validate the 'src' property.
+				if ( ! empty( $webfont['src'] ) ) {
+					foreach ( (array) $webfont['src'] as $src ) {
+						if ( empty( $src ) || ! is_string( $src ) ) {
+							trigger_error( __( 'Each webfont src must be a non-empty string.', 'gutenberg' ) );
+
+							return false;
+						}
+					}
+				}
+
+				// Check the font-weight.
+				if ( ! is_string( $webfont['font-weight'] ) && ! is_int( $webfont['font-weight'] ) ) {
+					trigger_error( __( 'Webfont font weight must be a properly formatted string or integer.', 'gutenberg' ) );
+
+					return false;
+				}
+
+				// Check the font-display.
+				if ( ! in_array( $webfont['font-display'], array( 'auto', 'block', 'fallback', 'swap' ), true ) ) {
+					$webfont['font-display'] = 'fallback';
+				}
+
+				$valid_props = array(
+					'ascend-override',
+					'descend-override',
+					'font-display',
+					'font-family',
+					'font-stretch',
+					'font-style',
+					'font-weight',
+					'font-variant',
+					'font-feature-settings',
+					'font-variation-settings',
+					'line-gap-override',
+					'size-adjust',
+					'src',
+					'unicode-range',
+				);
+
+				foreach ( $webfont as $prop => $value ) {
+					if ( ! in_array( $prop, $valid_props, true ) ) {
+						unset( $webfont[ $prop ] );
+					}
+				}
+
+				return $webfont;
+			}
+
+			/**
+			 * Generate and enqueue webfonts styles.
+			 *
+			 * @since 6.0.0
+			 */
+			public function generate_and_enqueue_styles() {
+				// Generate the styles.
+				$styles = $this->get_css();
+
+				// Bail out if there are no styles to enqueue.
+				if ( '' === $styles ) {
+					return;
+				}
+
+				// Enqueue the stylesheet.
+				wp_register_style( $this->stylesheet_handle, '' );
+				wp_enqueue_style( $this->stylesheet_handle );
+
+				// Add the styles to the stylesheet.
+				wp_add_inline_style( $this->stylesheet_handle, $styles );
+			}
+
+			/**
+			 * Generate and enqueue editor styles.
+			 *
+			 * @since 6.0.0
+			 */
+			public function generate_and_enqueue_editor_styles() {
+				// Generate the styles.
+				$styles = $this->get_css();
+
+				// Bail out if there are no styles to enqueue.
+				if ( '' === $styles ) {
+					return;
+				}
+
+				wp_add_inline_style( 'wp-block-library', $styles );
+			}
+
+			/**
+			 * Gets the `@font-face` CSS styles for locally-hosted font files.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @return string The `@font-face` CSS.
+			 */
+			public function get_css() {
+				$css = '';
+
+				foreach ( $this->webfonts as $webfont ) {
+					// Order the webfont's `src` items to optimize for browser support.
+					$webfont = $this->order_src( $webfont );
+
+					// Build the @font-face CSS for this webfont.
+					$css .= '@font-face{' . $this->build_font_face_css( $webfont ) . '}';
+				}
+
+				return $css;
+			}
+
+			/**
+			 * Order `src` items to optimize for browser support.
```

Comment:
```suggestion
			 * Orders 'src' items to optimize for browser support.
```
- Officially, markdown syntax is only valid in the body of a docblock, not the title.